src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -1779,6 +1779,70 @@ static Node* is_absolute( PhaseGVN *phase, PhiNode *phi_root, int true_path) {
   return x;
 }
 
+static Node* is_minmax(PhaseGVN* phase, PhiNode* phi_root, int true_path) {
+  assert(true_path != 0, "only diamond shape graph expected");
+
+  Node* region = phi_root->in(0);
+  IfNode* iff = region->in(1)->in(0)->as_If();
+  BoolNode* bol = iff->in(1)->as_Bool();
+  Node* cmp = bol->in(1);
```

Comment:
Could some of these values be const, or even TOP?

Reply:
I don't think these values can be TOP here as `PhiNode::is_diamond_phi()` and `RegionNode::is_diamond()`, which are called before this method is called, check to make sure that these nodes have the correct types and are not TOP.